### PR TITLE
the generate.sh exited when the cluster.yaml exists

### PR DESCRIPTION
**Pull request by hchenxa**
**What this PR does / why we need it**:

below code will exited when the cluster.yaml exists.
```
# Set up the output dir if it does not yet exist
mkdir -p $PWD/$OUTPUT
cp -n $PWD/cluster.yaml.template $PWD/$OUTPUT/cluster.yaml  >>>>>>>> The return code was 1
cp -n $PWD/machines.yaml.template $PWD/$OUTPUT/machines.yaml 
```

so in order to avoid the file overwrite and keep the updated file of cluster.yaml and machines.yaml, create this PR.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Specify your PR reviewers**:

/cc @gyliu513 @clyang82 

**Special notes for your reviewer**:

/sig ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by gyliu513** [comment]
/lgtm
/approve

You can post a PR to openstack provider as well to fix https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/master/cmd/clusterctl/examples/openstack/generate-yaml.sh#L143-L144

File: cmd/clusterctl/examples/ibmcloud/generate-yaml.sh [hunk 1]
```diff
@@ -112,8 +112,8 @@ cat "$WORKER_USER_DATA" > $USERDATA/$PROVIDER_OS/worker-user-data.sh
 
 # Set up the output dir if it does not yet exist
 mkdir -p $PWD/$OUTPUT
-cp -n $PWD/cluster.yaml.template $PWD/$OUTPUT/cluster.yaml
-cp -n $PWD/machines.yaml.template $PWD/$OUTPUT/machines.yaml
+cp -n $PWD/cluster.yaml.template $PWD/$OUTPUT/cluster.yaml || true
+cp -n $PWD/machines.yaml.template $PWD/$OUTPUT/machines.yaml || true
```
Comment (jichenjc) [inline cmd/clusterctl/examples/ibmcloud/generate-yaml.sh]: should we always overwrite the file?

Comment (gyliu513) [reply cmd/clusterctl/examples/ibmcloud/generate-yaml.sh]: I think we should not overwrite, as the overwrite will use template to overwrite which will clear some useful data.

Comment (hchenxa) [reply cmd/clusterctl/examples/ibmcloud/generate-yaml.sh]: the `cluster.yaml` and `machines.yaml` only contain some cluster info and machine info which should matainered by cluster deployer self, so I think we can only overwrite the `provider-components.yaml` when use overwrite flag.

cc @gyliu513 comments?

Comment (gyliu513) [reply cmd/clusterctl/examples/ibmcloud/generate-yaml.sh]: `provider-components.yaml` will always overwrite, this is the current logic already. I think the current logic is ok. @hchenxa @jichenjc

**Comment by xunpan** [comment]
FYI: this should be only MacOS issue. Linux works fine based on test.

**Comment by jichenjc** [comment]
ok, good to know :)

**Pull request merged**